### PR TITLE
add ibKubernetesEnabled config

### PR DESCRIPTION
**Pull request by zshi-redhat**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -200,11 +200,12 @@ echo 8 > /sys/class/net/ib0/device/sriov_numvfs
...
+* `pkey` (string, optional): InfiniBand pkey for VF, this field is used by [ib-kubernetes](https://www.github.com/Mellanox/ib-kubernetes) to add pkey with guid to InfiniBand subnet manager client e.g. [Mellanox UFM](https://www.mellanox.com/products/management-software/ufm), [OpenSM](https://docs.mellanox.com/display/MLNXOFEDv461000/OpenSM).
 * `ipam` (dictionary, optional): IPAM configuration to be used for this network, `dhcp` is not supported.
 * `link_state` (string, optional): Enforces link state for the VF. Allowed values: auto, enable, disable.
 * `rdmaIsolation` (boolean, optional): Enable RDMA network namespace isolation for RDMA workloads. More information
 about the system requirements to support this mode of operation can be found [here](https://github.com/Mellanox/rdma-cni)
+* `cniArgGUID` (bool, optional): Enforces ib-sriov-cni to get VF GUID from CNI-Args["mellanox.infiniband.app"], default to false. CNI-Args["mellanox.infiniband.app"] is configured by [ib-kubernetes](https://www.github.com/Mellanox/ib-kubernetes).
```
Comment (moshe010) [inline README.md:208]: working with ib-kubernetes we still allow to get GUID from CNI args. I think we need to change the name to IBKubernetesEnabled or something similar

File: cmd/ib-sriov-cni/main.go [hunk 2]
```diff
@@ -39,14 +39,18 @@ func init() {
 }
 
 func getGUIDFromConf(netConf *localtypes.NetConf) (string, error) {
+	// Take from CNI_ARGS if available
+	if netConf.CniArgGUID {
+		if guid, ok := netConf.Args.CNI["guid"]; ok {
```
Comment (moshe010) [inline cmd/ib-sriov-cni/main.go:44]: so you need to move the RuntimeConfig.InfinibandGUID  to here as well (Line 50). We will use this once InfinibandGUID   will be approved by the network-att-def same as mac see [1]

[1] https://docs.google.com/document/d/1V0R_MEiPapZOD_SuBjzTOdGYqILBFovuy1xacYCqFHg/edit

Comment (zshi-redhat) [reply cmd/ib-sriov-cni/main.go]: Done.
I thought with ib-kubernetes, it only allocates GUID for VF device, but it actually respect what's allocated in CNI_ARGS["guid"].

File: README.md (hunk 1 above)
Comment (zshi-redhat) [reply README.md]: Done

File: pkg/types/types.go [hunk 3]
```diff
@@ -13,21 +13,22 @@ import (
...
+	VFID                int
+	HostIFNames         string // VF netdevice name(s)
+	HostIFGUID          string // VF netdevice GUID
+	ContIFNames         string // VF names after in the container; used during deletion
+	GUID                string `json:"-"` // Taken from either CNI_ARGS "guid" attribute or from RuntimeConfig
+	PKey                string `json:"pkey"`
+	LinkState           string `json:"link_state,omitempty"` // auto|enable|disable
+	RdmaIso             bool   `json:"rdmaIsolation,omitempty"`
+	IBKubernetesEnabled bool   `json:"ibKubernetesEnabled,omitempty"`
+	RdmaNetState  rdmatypes.RdmaNetState
```
Comment (moshe010) [inline pkg/types/types.go:27]: gofmt error:
pkg/types/types.go:27: File is not `gofmt`-ed with `-s` (gofmt)
	RdmaNetState  rdmatypes.RdmaNetState
	Args         struct {
Makefile:74: recipe for target 'lint' failed
make: *** [lint] Error 1

Comment (zshi-redhat) [reply pkg/types/types.go]: Done

**Pull request merged**